sdk/core/azure-core/samples/shared_transport.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import os
+from azure.core.pipeline.transport import RequestsTransport
+from azure.storage.blob import BlobServiceClient
+
+connection_string = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+
+
+def shared_transport():
```

Comment:
Is it worth adding some kind of docstring describing what we're trying to achieve here?
e.g. "Listing containers across different Storage accounts, while using a single transport, and therefore sharing the connection pool between clients."

It's probably also worth adding a comment somewhere about what `session_owner` is, and what the implications of setting this to `True/False` are, and therefore why the transport should be in it's own context manager etc.